python/tox.ini

```diff
@@ -0,0 +1,117 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[tox]
+envlist = py37,py38,py39,linters
+
+[testenv]
+deps =
+    coverage
+    mock
+    nose
+    pytest
+setenv =
+    COVERAGE_FILE = test-reports/{envname}/.coverage
+    PYTEST_ADDOPTS = --junitxml=test-reports/{envname}/junit.xml -vv
+commands =
+    coverage run --source src --parallel-mode -m pytest {posargs}
+    coverage combine
+    coverage report -m
+    coverage html -d test-reports/{envname}/coverage-html
```

Comment:
Should we add a minimum coverage requirement here? I'm ok with either starting with something like `--fail-under 60` and then increasing as everything solidifies, or alternatively setting it very high from the get-go.

Reply:
This is a good idea. I add a threshold `90` to ensure a good coverage.